app/assets/javascripts/controllers/middleware_datasource/middleware_datasource_service.js

```diff
@@ -23,27 +26,98 @@ function MwAddDatasourceService($http) {
     {id: 'SYBASE', label: 'Sybase', name: 'SybaseDS', jndiName: 'java:jboss/datasources/SybaseDB',
       driverName: 'sybase', driverModuleName: 'com.sybase', driverClass: 'com.sybase.jdbc.SybDriver',
       connectionUrl: ':Tds:localhost:5000/mydatabase?JCONNECT_VERSION=6'},
-     {id: 'MARIADB', label: 'MariaDB', name: 'MariaDBDS', jndiName: 'java:jboss/datasources/MariaDBDS',
+    {id: 'MARIADB', label: 'MariaDB', name: 'MariaDBDS', jndiName: 'java:jboss/datasources/MariaDBDS',
        driverName: 'mariadb', driverModuleName: 'org.mariadb', driverClass: 'org.mariadb.jdbc.Driver',
        connectionUrl: '://localhost:3306/db_name'},
     {id: 'MYSQL', label: 'MySql', name: 'MySqlDS', jndiName: 'java:jboss/datasources/MySqlDS',
       driverName: 'mysql', driverModuleName: 'com.mysql', driverClass: 'com.mysql.jdbc.Driver',
-      connectionUrl: '://localhost:3306/db_name'}
+      connectionUrl: '://localhost:3306/db_name'},
   ];
+  var dsDriverNames = _.pluck(datasources, 'driverName');
 
-  this.getDatasources = function() {
+  self.getExistingJdbcDrivers = function(serverId) {
+    var deferred = $q.defer();
+    var BASE_URL = '/middleware_server/jdbc_drivers';
+    var parameterizedUrl = BASE_URL + '?server_id=' + serverId;
+
+    $http.get(parameterizedUrl).then(function(driverData) {
+      var transformedData = _.chain(driverData.data.data)
+        .filter(function(driver) {
+          return driver.properties['Driver Class'] !== null;
+        })
+        .map(function(driver) {
+          return {'id': driver.properties['Driver Name'].toUpperCase(),
+                  'label': driver.properties['Driver Name'],
+                  'xaDsClass': driver.properties['XA DS Class'],
+                  'driverClass': driver.properties['Driver Class']};
+      })
+      .value();
+
+      deferred.resolve(transformedData);
+    }).catch(function(errorMsg) {
+      deferred.reject(errorMsg);
+    });
+    return deferred.promise;
+  };
+
+  self.getDatasources = function() {
     return Object.freeze(datasources);
   };
 
-  this.determineConnectionUrl = function (dsSelection) {
-    var PREFIX = 'jdbc:';
+  self.isXaDriver = function(driver) {
+    return driver.hasOwnProperty('xaDsClass') && driver.xaDsClass !== '';
```

Comment:
Sure, I'll make that change in https://github.com/ManageIQ/manageiq-ui-classic/pull/149 since this is all tested and ready to go.